### PR TITLE
use url encoding instead base64

### PR DESCRIPTION
**Pull request by plally**

### CODE REVIEW

File: lua/cfc_http_restrictions/shared/filetypes/pls.lua [hunk 1]
```diff
@@ -39,7 +39,7 @@ end
 ---@param body string
 ---@return boolean
 function PLS.IsFileData( body )
-    if string.find( body, "%[playlist%]" ) then return true end
+    if string.find( body, "[playlist]", 1, true ) then return true end
```
Comment (brandonsturgeon) [inline lua/cfc_http_restrictions/shared/filetypes/pls.lua:42]: 👍

File: lua/cfc_http_restrictions/shared/url.lua [hunk 2]
```diff
@@ -34,10 +34,20 @@ function CFCHTTP.FindURLs( text )
     return urls
 end
 
+---@param url string
+---@return string
+function CFCHTTP.URLEncode( url )
+    url = url:gsub( "[^%w ]", function( c )
+        return string.format( "%%%02X", string.byte( c ) )
```
Comment (brandonsturgeon) [inline lua/cfc_http_restrictions/shared/url.lua:41]: I totally understand these formatting strings but could you include a quick comment summary of them, like, for other people?

**Review by brandonsturgeon — APPROVED** [review]
I didn't realize b64 wasn't always URL safe 👍

**Comment by Grocel** [comment]
> I didn't realize b64 wasn't always URL safe 👍

Yeah, that one is partially nasty. It can contain slashes (`/`) and pluses (`+`), which could cause some random seeming and hard to debug bugs.

I had the bless to debug such an issue in the past at a former work place, where my boss made the same mistake, but instead refused to fix it as would work "most of the time" and the fix being "to expensive". In fact fixing bugs he put into his code base and doing proper work was always "to expensive" to him, lol.

**Pull request merged**